src/components/Workspace/Workspace.tsx

```diff
@@ -63,10 +62,17 @@ export interface StatusData {
 export async function fetchProblemData(
   problemID: string
 ): Promise<ProblemData | null> {
-  const url = `${judgePrefix}/problem/${problemID}`;
-  const response = await fetch(url);
-  if (response.status !== 200) return null;
-  return await response.json();
+  const response = await fetch(
+    'https://raw.githubusercontent.com/cpinitiative/usaco-problems/main/problems.json'
+  );
+  const data = await response.json();
+  const problem = data[problemID];
+  const problemData = {
+    ...problem,
+    source: problem.source.sourceString,
+    title: problem.title.titleString,
+  };
+  return problemData;
```

Comment:
it would be more ideal if we downloaded the `problems.json` file and imported it locally, so we can save the time it takes to call `fetch()`. When the user visits `/usaco/1234`, they can't do anything until after this function returns, so minimizing latency here is really nice.

It's not a blocker though -- I think we should do this at some point, but I'm happy to merge this PR first and then improve it later.

Reply:
I feel like this fetch is relatively low-latency though, seeing as its not too large a json file...but I can definitely set something up to have it locally, like maybe another automated workflow similar to the one used in `usaco-guide` to auto-update problems. I did kind of like the idea of keeping all the problem metadata itself separate from the ide though, which is why I was more inclined to use fetch 😅

Reply:
I agree the json file isn't too large, but I think the latency is dominated by round-trip time / general connection overhead, not throughput (so even though the json file is small, there will still be some latency).

> maybe another automated workflow 

That sounds good to me! (was what I was thinking too)

> I did kind of like the idea of keeping all the problem metadata itself separate from the ide though

I agree that is nice :/ though we already have workflows for other repositories, so maybe adding another one here won't be too bad.

---

I still think automated workflow > fetch, but I agree that the latency is acceptable if we're lazy / it's not a big problem if we just use fetch.

Reply:
Oh, another thing; if we're hoping to support submissions on other sites (for example, CF) through usaco.guide ide, we'll certainly need fetch calls to fetch that data as well (we def cant have a database of all cf problems 😅 )

Reply:
oh that's a good point. ok if it's not that much slower I'm fine with just sticking to fetch() forever then. up to you!